docs/core/diagnostics/snippets/EventCounters/RequestEventSource.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Diagnostics.Tracing;
+
+public class RequestEventSource : EventSource
+{
+    public static readonly RequestEventSource Log = new RequestEventSource();
+
+    private IncrementingPollingCounter _requestRateCounter;
+    private int _requestCount = 0;
+
+    private RequestEventSource() =>
+        _requestRateCounter = new IncrementingPollingCounter("request-rate", this, () => _requestCount)
+        {
+            DisplayName = "Request Rate",
+            DisplayRateTimeScale = TimeSpan.FromSeconds(1)
+        };
+
+    public void AddRequest() => ++ _requestCount;
```

Comment:
If AddRequest() might be invoked on multiple threads then Interlocked.Increment() should be used instead of ++. If two threads run the ++ operator on the same variable at the same time it may increase only by 1 instead of 2, throwing off the count.

Reply:
Nevermind, I think you created this code explicitly so you could link to it from the concurrency section highlighting this issue ; )

Reply:
This is called out as part of the article, immediately following this it explains that. I could alternatively just include that as the source of truth to begin with?